src/did/src/transaction.rs

```diff
@@ -1021,6 +1021,21 @@ impl From<Bloom> for alloy::primitives::Bloom {
     }
 }
 
+impl Transaction {
+    /// This transaction is a special transaction that is used
+    /// for computing the PoW of the EVM.
+    pub fn pow_transaction() -> Self {
```

Comment:
The `Transaction` is a central struct in our architecture, I would not put this fn here. Please move it somewhere else as a top level fn.